crates/hyperswitch_connectors/src/connectors/jpmorgan/transformers.rs

```diff
@@ -142,24 +405,657 @@ impl<F, T> TryFrom<ResponseRouterData<F, JpmorganPaymentsResponse, T, PaymentsRe
+        let status = common_enums::AttemptStatus::from_transaction_state(transaction_state);
+
+        let transaction_id = item.response.transaction_id.clone();
+        let connector_response_reference_id = Some(transaction_id.clone());
+
+        let resource_id = ResponseId::ConnectorTransactionId(transaction_id.clone());
+
+        Ok(Self {
+            status,
+            response: Ok(PaymentsResponseData::TransactionResponse {
+                resource_id,
+                redirection_data: Box::new(None),
+                mandate_reference: Box::new(None),
+                connector_metadata: None,
+                network_txn_id: None,
+                connector_response_reference_id,
+                incremental_authorization_allowed: None,
+                charge_id: None,
+            }),
+            ..item.data
+        })
+    }
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct TransactionData {
+    payment_type: Option<String>,
+    status_code: i32,
+    txn_secret: Option<String>,
+    tid: Option<Secret<i64>>,
+    test_mode: Option<i8>,
+    status: Option<JpmorganTransactionStatus>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JpmorganRefundRequest {
+    pub merchant: MerchantRefundReq,
+    pub amount: MinorUnit,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MerchantRefundReq {
+    pub merchant_software: MerchantSoftware,
+}
+
+/*impl<F> TryFrom<&JpmorganRouterData<&RefundsRouterData<F>>> for JpmorganRefundRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &JpmorganRouterData<&RefundsRouterData<F>>) -> Result<Self, Self::Error> {
+        let merchant_software = MerchantSoftware {
+            company_name: String::from("JPMC"), //According to documentation, it should be the company name of software integrated to this API. If merchant is directly integrated, send "JPMC."
+            product_name: String::from("Hyperswitch"), //According to documentation, it should be the name of the product used for marketing purposes from a customer perspective. I. e. what the customer would recognize.
+            //https://developer.payments.jpmorgan.com/api/commerce/online-payments/online-payments#/operations/V2PaymentPost
+            version: Some(String::from("1.235")), //recheck, seek guidance
+        };
+
+        let merchant = MerchantRefundReq { merchant_software };
+
+        let amount = item.amount;
+
+        Ok(Self { merchant, amount })
+    }
+}*/
+
+impl<F> TryFrom<&JpmorganRouterData<&RefundsRouterData<F>>> for JpmorganRefundRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(_item: &JpmorganRouterData<&RefundsRouterData<F>>) -> Result<Self, Self::Error> {
+        Err(errors::ConnectorError::NotImplemented("Refunds".to_string()).into())
+    }
+}
```

Comment:
Add this error in `get_url` function itself

Reply:
alright